Modules/Core/Common/include/itkMath.h

```diff
@@ -318,25 +319,26 @@ FloatAlmostEqual(T                                        x1,
     return true;
   }
 
-#if defined(__APPLE__) && (__clang_major__ == 3) && (__clang_minor__ == 0) && defined(NDEBUG) && defined(__x86_64__)
-  Detail::FloatIEEE<T> x1f(x1);
-  Detail::FloatIEEE<T> x2f(x2);
-  double               x1fAsULP = static_cast<double>(x1f.AsULP());
-  double               x2fAsULP = static_cast<double>(x2f.AsULP());
-  double               ulps = x1fAsULP - x2fAsULP;
-  if (ulps < 0)
+  // This check for different signs is necessary for several reasons, see the blog link above.
+  // Subtracting the signed-magnitude representation of floats using twos-complement
+  // math isn't particularly meaningful, and the subtraction would produce a 33-bit
+  // result and overflow an int.
+  if (std::signbit(x1) != std::signbit(x2))
   {
-    ulps = -ulps;
+    // Check for equality to make sure +0 == -0
+    if (x1 == x2)
+    {
+      return true;
+    }
+    return false;
```

Comment:
Interesting issue @seanm By the way, it looks like these five lines of code are equivalent to just `return  x1 == x2`. In this case I would prefer one line of code, instead on five!

Reply:
Indeed. I blindly copied https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/

Will change.